### PR TITLE
Added inactive state for checkbox

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR add a field-transformer for boolean values and an inactive state to the checkbox and switch component.

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/Switch.js [hunk 1]
```diff
@@ -32,10 +33,14 @@ export default class Switch extends React.PureComponent<Props> {
...
             className,
+            inactive,
         } = this.props;
         const switchClass = classNames(
             switchStyles.switch,
-            className
+            className,
+            {
+                [switchStyles.inactive]: inactive,
+            }
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/Switch.js]: if inactive I would set `disabled` or `readonly` flag on the input.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ import checkboxStyles from './checkbox.scss';
 type Props = SwitchProps & {
     skin: 'dark' | 'light',
     className?: string,
+    inactive?: boolean,
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js]: How about `active` as a mandatory parameter, and setting it to `true` by default? Positive names are easier to understand IMO.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js [hunk 3]
```diff
@@ -27,10 +28,14 @@ export default class Checkbox extends React.PureComponent<Props> {
...
             children,
             className,
+            inactive,
         } = this.props;
         const checkboxClass = classNames(
             checkboxStyles.checkbox,
-            checkboxStyles[skin],
+            {
+                [checkboxStyles.inactive]: inactive,
+                [checkboxStyles[skin]]: !inactive,
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js]: Any reason not to set the skin when the button is not active? Maybe there will even be different inactive styles for each skin at some point.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/Switch.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ type Props = SwitchProps & {
     icon?: string,
     type: string,
     onChange?: (checked: boolean, value?: string | number) => void,
+    inactive?: boolean,
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/Switch.js]: Would make the same change as in the `Checkbox` here

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js (hunk 3 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js]: currently the same css attributes are set differently. so there is currently no difference - but we can set both classes.

**Comment by wachterjohannes** [comment]
@danrot @alexander-schranz should be good now

**Pull request merged**